openpype/plugins/publish/collect_anatomy_context_data.py

```diff
@@ -92,5 +92,13 @@ def process(self, context):
                 }
             })
 
+        # todo: some code actually expects the dict itself and others doesn't
+        #       question: what should it be?
+        intent = context.data.get("intent")
+        if intent and isinstance(intent, dict):
+            intent = intent.get("value")
+        if intent:
+            context_data["intent"] = intent
+
```

Comment:
Intent should be a dictionary with "value" and "label", to be able tell if you want use value or label of the intent in templates.
```suggestion
```

Reply:
Removed with recent commit. However, this logic with the if statement still feels like we should somehow correct other logic elsewhere so that we know it's always a `dict` to begin with.